tests/test_lattice.py

```diff
@@ -1,12 +1,23 @@
 from unittest.mock import patch
+import time
 import logging
 
-# import time
-
 import matplotlib
 
-matplotlib.use("Agg")
+try:
+    import jax.numpy as jnp
+except ImportError:
+    jnp = None
+try:
+    import tensorflow as tf
+except ImportError:
+    tf = None
+try:
+    import torch
+except ImportError:
+    torch = None
```

Comment:
no need to do all these, as pytest fixture tfb, jaxb etc can automatically skip the test without installation of the corresponding packages